### PR TITLE
Translating ch06-03 in French.

### LINKED ISSUE #60: Translating ch06-03
I'm working on the French translation of this page.

### PR DESCRIPTION
**Pull request by Jimskapt**
Closes #60.

### CODE REVIEW

File: FRENCH/src/translation-terms.md [hunk 1]
```diff
@@ -43,6 +44,7 @@ français.
 | dependency | dépendance | - |
 | destructure | déstructurer | - |
 | DevOps | DevOps | - |
+| directory | répertoire | - |
```
Comment (LeSeulArtichaut) [inline FRENCH/src/translation-terms.md]: `dossier` is probably more used
```suggestion
| directory | dossier | - |
```

File: FRENCH/src/ch06-03-if-let.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+announcing the state of the quarters, we could do that with a `match`
+expression like this:
+-->
+
+Nous pouvons joindre un `else` à un `if let`. Le bloc de code qui va dans le
+`else` est le même que le bloc de code qui va dans le cas `_` dans l'expression
+`match`. Souvenez-vous de la définition de l'énumération `USACoin` de l'encart
+6-4, où la variante `Quarter` avait aussi une valeur `USAState`. Si nous
+voulions compter toutes les pièces qui ne sont pas des `Quarter` que nous voyons
+passer, tout en signalant l'état des `Quarter`, nous pourrions faire ceci avec
```
Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-03-if-let.md]:
```suggestion
passer, tout en affichant l'état des `Quarter`, nous pourrions faire ceci avec
```

Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-03-if-let.md]:
```suggestion
passer, tout en signalant l'état des `Quarter`, nous pourrions le faire avec
```

File: FRENCH/src/ch06-03-if-let.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+    USACoin::Quarter(etat) => println!("Il s'agit d'un Quarter de l'état de {:?} !", etat),
+    _ => compteur += 1,
+}
+```
+
+<!--
+Or we could use an `if let` and `else` expression like this:
+-->
+
+Ou nous pourrions utiliser des expressions `if let` et `else` comme ceci :
```
Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-03-if-let.md]:
```suggestion
Ou nous pourrions utiliser une expression `if let`/`else` comme ceci :
```

File: FRENCH/src/ch06-03-if-let.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+}
+```
+
+<!--
+If you have a situation in which your program has logic that is too verbose to
+express using a `match`, remember that `if let` is in your Rust toolbox as well.
+-->
+
+Si vous vous trouvez dans une situation dans laquelle votre programme a une
+logique qui est trop verbeuse en utilisant un `match`, souvenez-vous que
```
Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-03-if-let.md]: Translation is too literal. Suggestion:
```markdown
Si vous vous trouvez que votre programme est trop lourd
en utilisant un `match`, souvenez-vous que [...]
```

File: FRENCH/src/ch06-03-if-let.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+type helps you use the type system to prevent errors. When enum values have
+data inside them, you can use `match` or `if let` to extract and use those
+values, depending on how many cases you need to handle.
+-->
+
+Nous avons désormais appris comment utiliser les énumérations pour créer des
+types personnalisés qui peuvent faire partie d'un jeu de valeurs recensées. Nous
+avons montré comment le type `Option<T>` de la bibliothèque standard vous aide
+à utiliser le système de types pour éviter les erreurs. Lorsque les valeurs
+d'énumération ont une valeur à l'intérieur d'elles, vous pouvez utiliser `match`
```
Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-03-if-let.md]:
```suggestion
d'énumération contiennent des données, vous pouvez utiliser `match`
```

File: FRENCH/src/ch06-03-if-let.md [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+nombre de cas que vous voulez gérer.
+
+<!--
+Your Rust programs can now express concepts in your domain using structs and
+enums. Creating custom types to use in your API ensures type safety: the
+compiler will make certain your functions get only values of the type each
+function expects.
+-->
+
+Vos programmes Rust peuvent maintenant décrire des concepts métier à l'aide de
```
Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-03-if-let.md]: Is `métier` really useful here?
```suggestion
Vos programmes Rust peuvent maintenant décrire des concepts à l'aide de
```

File: FRENCH/src/ch06-03-if-let.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+structures et d'énumérations. Créer des types personnalisés à utiliser dans
+votre API assure la sécurité des types : le compilateur s'assurera que vos
+fonctions ne reçoivent uniquement des valeurs du type attendu.
```
Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-03-if-let.md]:
```suggestion
fonctions reçoivent uniquement des valeurs du type attendu.
```
Or
```suggestion
fonctions ne reçoivent que des valeurs du type attendu.
```

File: FRENCH/src/translation-terms.md [hunk 8]
```diff
@@ -15,6 +15,7 @@ français.
 | associated function | fonction associée | - |
 | binary crate | crate binaire | s'utilise au féminin |
 | *n*-bit number | nombre encodé sur *n* bits | - |
+| boilerplate code | code standard | - |
```
Comment (LeSeulArtichaut) [inline FRENCH/src/translation-terms.md]: May not need a particular fixed translation, and should be reajusted depending on the context instead

File: FRENCH/src/translation-terms.md [hunk 9]
```diff
@@ -54,6 +56,7 @@ français.
 | floating-point number | nombre à virgule flottante | - |
 | function | fonction | - |
 | generics | génériques / généricité | - |
+| generic type parameter | paramètre de type générique | - |
```
Comment (LeSeulArtichaut) [inline FRENCH/src/translation-terms.md]: Not used in this chapter, shouldn't be part of this merge

File: FRENCH/src/translation-terms.md [hunk 10]
```diff
@@ -78,6 +81,7 @@ français.
 | library crate | crate de bibliothèque | s'utilise au féminin |
 | lifetime | durée de vie | - |
 | linker | linker | - |
+| literal value | valeur pure | - |
```
Comment (LeSeulArtichaut) [inline FRENCH/src/translation-terms.md]: Already discussed in #66, should be changed here

File: FRENCH/src/translation-terms.md [hunk 11]
```diff
@@ -86,8 +90,10 @@ français.
 | memory management | gestion de mémoire | - |
 | method | méthode | - |
 | module | module | - |
+| module system | système de modules | - |
```
Comment (LeSeulArtichaut) [inline FRENCH/src/translation-terms.md]: Not used in this chapter

File: FRENCH/src/translation-terms.md [hunk 12]
```diff
@@ -86,8 +90,10 @@ français.
 | memory management | gestion de mémoire | - |
 | method | méthode | - |
 | module | module | - |
+| module system | système de modules | - |
 | mutability | mutabilité | - |
 | mutable | mutable | modifiable |
+| namespace | espace de nom | - |
```
Comment (LeSeulArtichaut) [inline FRENCH/src/translation-terms.md]: Not used in this chapter

File: FRENCH/src/translation-terms.md [hunk 13]
```diff
@@ -100,6 +106,7 @@ français.
 | pattern-matching | filtrage par motif | - |
 | placeholder | espace réservé | `{}` pour `fmt` |
 | prelude | étape préliminaire | - |
+| privacy | protection | explication de la protection des éléments d'un module |
```
Comment (LeSeulArtichaut) [inline FRENCH/src/translation-terms.md]: Not used in this chapter

File: FRENCH/src/translation-terms.md [hunk 14]
```diff
@@ -138,11 +145,13 @@ français.
 | `String` | `String` | nom féminin (une `String`) |
 | struct | structure | - |
 | submodule | sous-module | - |
+| syntax sugar | sucre syntaxique | - |
 | systems concept | notion système | - |
 | systems-level | niveau système | - |
 | systems-level code | code système | - |
 | terminal | terminal | - |
 | trait | trait | - |
+| tree | arborescence | - |
```
Comment (LeSeulArtichaut) [inline FRENCH/src/translation-terms.md]: Not used in this chapter

File: FRENCH/src/translation-terms.md [hunk 15]
```diff
@@ -160,4 +169,5 @@ français.
 | vector | vecteur | - |
 | version control system (VCS) | système de gestion de versions (VCS) | - |
 | warning | avertissement | - |
+| wildcard | jocker | - |
```
Comment (LeSeulArtichaut) [inline FRENCH/src/translation-terms.md]: Not used in this chapter

File: FRENCH/src/SUMMARY.md [hunk 16]
```diff
@@ -13,3 +13,5 @@
     - [Hello, Cargo!](ch01-03-hello-cargo.md)
 
 - [Programmer un jeu de devinettes](ch02-00-guessing-game-tutorial.md)
+
+- [Une structure de contrôle concise avec `if let`](ch06-03-if-let.md)
```
Comment (LeSeulArtichaut) [inline FRENCH/src/SUMMARY.md]: `avec` should be used when referring to the concept, not the structure itself
```suggestion
- [Une structure de contrôle concise: `if let`](ch06-03-if-let.md)
```
Or
```suggestion
- [Contrôle de flot concis avec `if let`](ch06-03-if-let.md)
```

File: FRENCH/src/ch06-03-if-let.md [hunk 17]
```diff
@@ -0,0 +1,279 @@
+<!--
+## Concise Control Flow with `if let`
+-->
+
+## Une structure de contrôle concise avec `if let`
```
Comment (LeSeulArtichaut) [inline FRENCH/src/ch06-03-if-let.md]: Should be changed to whatever was decided in the SUMMARY.md

File: FRENCH/src/translation-terms.md (hunk 9 above)
Comment (Jimskapt) [reply FRENCH/src/translation-terms.md]: Yes, I remember I have done this mistake only one time, a long time ago, it's probably here.
I don't think it is very important, so please do several separated Push Request on concerned chapters if you want to do this work :wink:

File: FRENCH/src/ch06-03-if-let.md (hunk 6 above)
Comment (Jimskapt) [reply FRENCH/src/ch06-03-if-let.md]: I don't think so, but the (english) authors has said so.
Probably to emphasise that is a tool that end-developpers can use ?

File: FRENCH/src/translation-terms.md (hunk 8 above)
Comment (Jimskapt) [reply FRENCH/src/translation-terms.md]: Of course !

File: FRENCH/src/ch06-03-if-let.md (hunk 6 above)
Comment (LeSeulArtichaut) [reply FRENCH/src/ch06-03-if-let.md]: Maybe use `concrets` then?
```suggestion
Vos programmes Rust peuvent maintenant décrire des concepts concrets à l'aide de
```

**Comment by LeSeulArtichaut** [comment]
@Jimskapt You should rebase `FR-ch06-03` onto `french-release` in order to resolve the merge conflicts

**Comment by Jimskapt** [comment]
@LeSeulArtichaut  I never used this tool, I learned how it works thanks to [this video](https://www.youtube.com/watch?v=CRlGDDprdOQ). I've locally tried to `git rebase FR-ch06-03` on the `french-release` branch, but I've got a lot of successive "fusion errors", which are error-prone (I stopped it on the 4th). They come essentially from `FRENCH/src/SUMMARY.md` and `FRENCH/src/translation-terms.md` files. So now I think it is easier and less dangerous to do one simple merge commit directly in GitHub than rebase.

**Pull request merged**